puppet/chassis.rb

```diff
@@ -177,7 +177,25 @@ def self.install_extension(extension)
 			repo = extension
 		end
 
-		folder = @@dir + '/extensions/' + extension.split('/').last.gsub(/.git$/, '').downcase
+		# We need to add a version to allow for a seamless upgrade to Bionic Beaver.
+		if config["version"] >= 3
+			# Check for existing extensions that have a hyphen and remove them.
+			old_extensions = Dir.glob(@@dir + '/extensions/*')
+			old_extensions.each { |extension|
+				next unless extension.include? "-"
+					# Delete the old extension if it has a hyphen in it.
+					FileUtils.remove_dir(extension)
+					new_extension = extension.gsub(/-/, '_')
+					puts "We've renamed #{extension} to #{new_extension} as hyphens aren't allowed in Puppet: https://puppet.com/docs/puppet/5.4/lang_reserved.html#classes-and-defined-resource-types"
+					puts "If you have #{extension} in one of your yaml configuration files we would recommend altering it to #{new_extension}."
```

Comment:
Might want to move this outside of the loop so it's only done once? Also, I'd say we can just stick to the first sentence.

Can we handle the config bit ourselves in the config normalisation step? Maybe warn there instead.

Reply:
Done and done!